src/ImageSharp/Processing/Processors/Convolution/MedianRowOperation{TPixel}.cs

```diff
@@ -0,0 +1,164 @@
+        private readonly Rectangle bounds;
+        private readonly Buffer2D<TPixel> targetPixels;
+        private readonly Buffer2D<TPixel> sourcePixels;
+        private readonly KernelSamplingMap map;
+        private readonly int kernelSize;
+        private readonly bool preserveAlpha;
+
+        public MedianRowOperation(Rectangle bounds, Buffer2D<TPixel> targetPixels, Buffer2D<TPixel> sourcePixels, KernelSamplingMap map, int kernelSize, Configuration configuration, bool preserveAlpha)
+        {
+            this.bounds = bounds;
+            this.configuration = configuration;
+            this.targetPixels = targetPixels;
+            this.sourcePixels = sourcePixels;
+            this.map = map;
+            this.kernelSize = kernelSize;
+            this.preserveAlpha = preserveAlpha;
+            int kernelCount = this.kernelSize * this.kernelSize;
+            this.yChannelStart = kernelCount;
+            this.zChannelStart = this.yChannelStart + kernelCount;
+            this.wChannelStart = this.zChannelStart + kernelCount;
+        }
+
+        public void Invoke(int y, Span<Vector4> span)
+        {
+            // Span has kernelSize^2 followed by bound width.
+            int boundsLeft = this.bounds.Left;
+            int boundsWidth = this.bounds.Width;
+            int boundsRight = this.bounds.Right;
+            int kernelCount = this.kernelSize * this.kernelSize;
+            Span<Vector4> kernelBuffer = span.Slice(0, kernelCount);
+            Span<Vector4> channelVectorBuffer = span.Slice(kernelCount, kernelCount);
+            Span<Vector4> targetBuffer = span.Slice(kernelCount << 1, boundsWidth);
+
+            // Stack 4 channels of floats in the space of Vector4's.
+            Span<float> channelBuffer = MemoryMarshal.Cast<Vector4, float>(channelVectorBuffer);
+            var xChannel = channelBuffer.Slice(0, kernelCount);
+            var yChannel = channelBuffer.Slice(this.yChannelStart, kernelCount);
+            var zChannel = channelBuffer.Slice(this.zChannelStart, kernelCount);
+
+            var xOffsets = this.map.GetColumnOffsetSpan();
+            var yOffsets = this.map.GetRowOffsetSpan();
+            var baseXOffsetIndex = 0;
+            var baseYOffsetIndex = (y - this.bounds.Top) * this.kernelSize;
+
+            if (this.preserveAlpha)
+            {
+                for (var x = boundsLeft; x < boundsRight; x++)
+                {
+                    var index = 0;
+                    for (var w = 0; w < this.kernelSize; w++)
+                    {
+                        var j = yOffsets[baseYOffsetIndex + w];
+                        var row = this.sourcePixels.DangerousGetRowSpan(j);
+                        for (var z = 0; z < this.kernelSize; z++)
+                        {
+                            var k = xOffsets[baseXOffsetIndex + z];
+                            var pixel = row[k];
+                            kernelBuffer[index + z] = pixel.ToVector4();
```

Comment:
Because the `z` loops runs over neighboring rows, this buffer would need to contain at least several rows of the source image. 

Do you feel that is worth the memory?

Reply:
We can still read/convert the source row slice one at a time though if you're sampling one kernel row at a time?

Reply:
Please see in the comments below how this is done.